omnigibson/objects/controllable_object.py

```diff
@@ -529,20 +536,35 @@ def get_control_dict(self):
                 - cc_force: (n_dof,) per-joint centripetal and centrifugal forces
         """
+        # Note that everything here uses the ControllableObjectViewAPI because these are faster implementations of
+        # the functions that this class also implements. The API centralizes access for all of the robots in the scene
+        # removing the need for multiple reads and writes.
+        # TODO(cgokmen): CachedFunctions can now be entirely removed since the ControllableObjectViewAPI already implements caching.
         fcns = CachedFunctions()
-        fcns["_root_pos_quat"] = self.get_position_orientation
+        fcns["_root_pos_quat"] = lambda: ControllableObjectViewAPI.get_position_orientation(self.articulation_root_path)
         fcns["root_pos"] = lambda: fcns["_root_pos_quat"][0]
         fcns["root_quat"] = lambda: fcns["_root_pos_quat"][1]
-        fcns["root_lin_vel"] = self.get_linear_velocity
-        fcns["root_ang_vel"] = self.get_angular_velocity
-        fcns["root_rel_lin_vel"] = self.get_relative_linear_velocity
-        fcns["root_rel_ang_vel"] = self.get_relative_angular_velocity
-        fcns["joint_position"] = lambda: self.get_joint_positions(normalized=False)
-        fcns["joint_velocity"] = lambda: self.get_joint_velocities(normalized=False)
-        fcns["joint_effort"] = lambda: self.get_joint_efforts(normalized=False)
-        fcns["mass_matrix"] = lambda: self.get_mass_matrix(clone=False)
-        fcns["gravity_force"] = lambda: self.get_generalized_gravity_forces(clone=False)
-        fcns["cc_force"] = lambda: self.get_coriolis_and_centrifugal_forces(clone=False)
+        fcns["root_lin_vel"] = lambda: ControllableObjectViewAPI.get_linear_velocity(self.articulation_root_path)
+        fcns["root_ang_vel"] = lambda: ControllableObjectViewAPI.get_angular_velocity(self.articulation_root_path)
+        fcns["root_rel_lin_vel"] = lambda: ControllableObjectViewAPI.get_relative_linear_velocity(
+            self.articulation_root_path
+        )
+        fcns["root_rel_ang_vel"] = lambda: ControllableObjectViewAPI.get_relative_angular_velocity(
+            self.articulation_root_path
+        )
+        fcns["joint_position"] = lambda: ControllableObjectViewAPI.get_joint_positions(self.articulation_root_path)
+        fcns["joint_velocity"] = lambda: ControllableObjectViewAPI.get_joint_velocities(self.articulation_root_path)
+        fcns["joint_effort"] = lambda: ControllableObjectViewAPI.get_joint_efforts(self.articulation_root_path)
+        fcns["mass_matrix"] = lambda: ControllableObjectViewAPI.get_mass_matrix(self.articulation_root_path)
+        # TODO: Move gravity force computation dummy to this class instead of BaseRobot
+        fcns["gravity_force"] = lambda: (
+            ControllableObjectViewAPI.get_generalized_gravity_forces(self.articulation_root_path)
+            if not self.fixed_base
+            else DummyObjectViewAPI.get_generalized_gravity_forces(self._dummy.articulation_root_path)
```

Comment:
note to myself: what is `DummyObjectViewAPI`?

Reply:
pretty sure it's for the dummy duplicate object associated with a given non-fixed-base controllable object, e.g.: mobile manipulator?

Reply:
Yup. It's a version of the ControllableObjectViewAPI that only reads the dummies.